inference-engine/src/mkldnn_plugin/nodes/mkldnn_pad_node.cpp

```diff
@@ -0,0 +1,254 @@
+
+    parallel_nt(0, [&](const int ithr, const int nthr) {
+        size_t start = 0, end = 0;
+        SizeVector counters(dimsSize_1, 0);
+        splitter(workAmountDst, nthr, ithr, start, end);
+
+        parallel_init(start, dimsSize_1, counters, dstDims);
+        int dstIdx = 0;
+        for (size_t i = 0; i < dimsSize_1; ++i)
+            dstIdx += counters[i] * params.dstStrides[i];
+
+        for (size_t iwork = start; iwork < end; ++iwork, dstIdx += inputSV) {
+            if (!isEdge) {
+                size_t j;
+                for (j = 0; j < dimsSize_1; ++j) {
+                    if ((counters[j] < padsBegin[j]) || (counters[j] >= params.srcODims[j]))
+                        break;
+                }
+
+                if (j != dimsSize_1) {
+                    std::fill_n(&dstData[dstIdx], dstDims[dimsSize_1], padValue);
+                    parallel_step(dimsSize_1, counters, dstDims);
+                    continue;
+                }
+            }
+
+            int srcIdx = 0;
+            for (size_t i = 0; i < dimsSize_1; ++i) {
+                int idx = (counters[i] < padsBegin[i]) ? 0 :
+                          ((counters[i] >= params.srcODims[i]) ? (srcDims[i] - 1) : (counters[i] - padsBegin[i]));
+                srcIdx += idx * params.srcStrides[i];
+            }
+
+            std::fill_n(&dstData[dstIdx], padsBegin[dimsSize_1],
+                        isEdge ? srcData[srcIdx] : padValue);
+            cpu_memcpy(&dstData[dstIdx + padsBegin[dimsSize_1]], &srcData[srcIdx],
+                       srcDims[dimsSize_1] * sizeof(float));
+            std::fill_n(&dstData[dstIdx + params.srcODims[dimsSize_1]], padsEnd[dimsSize_1],
+                        isEdge ? srcData[srcIdx + srcDims[dimsSize_1] - 1] : padValue);
+
+            parallel_step(dimsSize_1, counters, dstDims);
+        }
+    });
+}
+
+void MKLDNNPadNode::padReflectOrSymmetric(const float *srcData, float* dstData, SizeVector srcDims, SizeVector dstDims,
+        const bool isSymmetric) {
+    SizeVector src_2;
+    for (size_t i = 0; i < srcDims.size(); i++)
+        src_2.push_back(srcDims[i] + params.srcODims[i] - 2 + isSymmetric);
+
+    size_t dimsSize_1 = dstDims.size() - 1;
+    size_t inputSV = dstDims[dimsSize_1];
+    size_t workAmountDst = getWorkAmountDst();
+
+    parallel_nt(0, [&](const int ithr, const int nthr) {
+        size_t start = 0, end = 0;
+        SizeVector counters(dimsSize_1, 0);
+        splitter(workAmountDst, nthr, ithr, start, end);
+
+        parallel_init(start, dimsSize_1, counters, dstDims);
+        int dstIdx = 0;
+        for (size_t i = 0; i < dimsSize_1; ++i)
+            dstIdx += counters[i] * params.dstStrides[i];
+
+        for (size_t iwork = start; iwork < end; ++iwork, dstIdx += inputSV) {
+            int srcIdx = 0;
+            for (size_t i = 0; i < dimsSize_1; ++i) {
+                int idx = (counters[i] < padsBegin[i]) ? (padsBegin[i] - counters[i] - isSymmetric) :
```

Comment:
Are you sure that isSymmetric will not be implicitly casted to int here on each iteration? I would prefer to have separate variable which will equal to `0` or `1` depending on isSymmetric value,

Reply:
Added separate variable for it, thank you